pkg/codegen/nodejs/gen.go

```diff
@@ -1184,6 +1228,15 @@ func (mod *modContext) genFunctionOutputVersion(
 
 	originalName := tokenToFunctionName(fun.Token)
 	fnOutput := fmt.Sprintf("%sOutput", originalName)
+	returnType := "void"
+	if fun.ReturnType != nil {
+		if objectType, ok := fun.ReturnType.(*schema.ObjectType); ok && objectType != nil {
```

Comment:
What if the return type is an `"$ref"` based object? We don't want to generate a special type there. I suspect we will need a `fun.GenerateReturnType() bool` to discriminate referenced objects from inline objects.